test/functional/eval/wait_spec.lua

```diff
@@ -0,0 +1,40 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear, eval, eq, feed, feed_command, next_msg, nvim, source =
+helpers.clear, helpers.eval, helpers.eq, helpers.feed, helpers.feed_command,
+helpers.next_msg, helpers.nvim, helpers.source
+
+before_each(function()
+  clear()
+  local channel = nvim('get_api_info')[1]
+  nvim('set_var', 'channel', channel)
+end)
+
+describe('wait()', function()
+  it('waits and returns 0 when condition is satisfied', function()
+    source([[
+    let g:_awake = 0
+    call timer_start(1000, { -> nvim_command('let g:_awake = 1') })
+    ]])
+    eq(0, eval('wait(1500, { -> g:_awake })'))
+    eq(1, eval('g:_awake'))
+  end)
+
+  it('returns -1 on timeout', function()
+    eq(-1, eval('wait(0, "0")'))
+  end)
+
+  it('returns -2 when interrupted', function()
+    feed_command('call rpcnotify(g:channel, "ready") | '..
```

Comment:
`command` is preferred over `feed_command` unless there's a specific reason to use `feed_command`. 
 `command` sends a command directly, whereas `feed_command` is equivalent to a user typing `:...<cr>`.

Reply:
Yeah I used the asynchronous one because I want to call `feed('<c-c')` during the wait